### PR TITLE
Added Light and Dark Grey Main and Secondary themes as per Request #2600

### LINKED ISSUE #2600: [Feature Request]: Grey option for main and secondary color theme
### Guidelines

- [X] I have searched the [issue tracker](https://github.com/FreeTubeApp/FreeTube/issues) for a feature request that matches the one I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.

### Problem Description

Can you add a Light-Dark Grey color option for Main and Secondary color theme ?

### Proposed Solution

Can you add a Light-Dark Grey color option for Main and Secondary color theme ?

### Alternatives Considered

Can you add a Light-Dark Grey color option for Main and Secondary color theme ?

### Issue Labels

new feature, new optional setting

### Additional Information

Here are some color sample suggestions:

Davy’s Gray
Hex #555555
RGB 85, 85, 85
CMYK 0, 0, 0, 67

Dark Gray
Hex #A9A9A9
RGB 169, 169, 169
CMYK 0, 0, 0, 34

Battleship Gray
Hex #848482
RGB 132, 132, 130
CMYK 0, 0, 2, 48

Gray
Hex #808080
RGB 128, 128, 128
CMYK 0, 0, 0, 50

Dim Gray
Hex #696969
RGB 105, 105, 105
CMYK 0, 0, 0, 59

### PR DESCRIPTION
**Pull request by fdarcey**
# Title

Added Light and Dark Grey Main and Secondary themes as per Request #2600

## Pull Request Type
- [ ] Bugfix
- [x] Feature Implementation
- [ ] Documentation
- [ ] Other

## Related issue

closes #2600

## Description

FreeTube/src/renderer/themes.ccs now has:
- .mainLightGrey with primary-color: #B1B1B1, primary-color-hover: #848482, and primary-color-active: #7E7E7E. Lines 266, 365-369.
- .mainDarkGrey with primary-color: ##696969, primary-color-hover: #555555, and primary-color-active: #4E4E4E. Lines 267, 371-375.
- .secLightGrey with primary-color-active: #4E4E4E, accent-color-hover: #848482, accent-color-active: #7E7E7E, 
accent-color-light: #cfcfcf, accent-color-visited: #797979, accent-color-opacity1: rgba(177,177,177,0.04), with accent-color-opacity2, 3, and 4 having their fourth value at 0.12, 0.16, and 0.24, respectively. Lines 561, 756-766.
- .secDarkGrey with accent-color: #696969, accent-color-hover: #555555, accent-color-active: #4E4E4E, 
accent-color-light: #A5A5A5, accent-color-visited: #444444, accent-color-opacity1: rgba(105,105,105,0.04), with accent-color-opacity2, 3, and 4 having their fourth value at 0.12, 0.16, and 0.24, respectively. Lines 562, 768-778.

FreeTube/src/renderer/helpers/colors.js has:
- { name: 'LightGrey', value: '#B1B1B1'}, and { name: 'DarkGrey', value: '#696969'} on line 18 and 19 under export const colors.

FreeTube/static/locales/en-US.yaml has:
- Light Grey: Light Grey and Dark Grey: Dark Grey on line 230 and 231.

Overall, the user could now select Light Grey and Dark Grey for main and secondary themes and have the corresponding UI elements switch to those colors.
Note: I have only updated en-US.yalm with Light and Dark Grey, deciding to allow the users more experienced with translations add in such for other languages. This means that until then, the main and secondary color menus will show them with "Setting.Theme Settings." preceding them.

## Screenshots 

image

## Testing 

This has been tested, and the result is shown in the screenshot.

## Desktop

- **OS:** Windows
- **OS Version:** 11
- **FreeTube version:** 0.19.1 Beta

### CODE REVIEW

**Comment by absidue** [comment]
Issues are considered closed when the feature has been implemented or bug fixed. If we waited for all translators to log into weblate and translate stuff before closing issues, we would have them open for months or longer.

File: src/renderer/themes.css [hunk 1]
```diff
@@ -263,6 +263,9 @@ it can be safely elided. This looks quite pleasant on this theme. */
   --logo-text-bar-color: url("../../_icons/textBlackSmall.png");
 }
 
+.mainLightGrey,
+.mainDarkGrey,
```
Comment (ChunkyProgrammer) [inline src/renderer/themes.css:267]: Looks like these 3 lines (266-268) can be removed

File: static/locales/en-US.yaml [hunk 2]
```diff
@@ -227,6 +227,8 @@ Settings:
       Amber: Amber
       Orange: Orange
       Deep Orange: Deep Orange
+      Light Grey: Light Grey
+      Dark Grey: Dark Grey
```
Comment (ChunkyProgrammer) [inline static/locales/en-US.yaml:231]: I believe it's spelled "Gray" in en-US

File: src/renderer/themes.css (hunk 1 above)
Comment (fdarcey) [reply src/renderer/themes.css]: Removed the excess code you pointed out, as well as the same for the sec greys. My testing revealed that they still worked without them.

File: static/locales/en-US.yaml (hunk 2 above)
Comment (fdarcey) [reply static/locales/en-US.yaml]: Grey changed to Gray.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
I would suggest changing everything to gray instead of grey just to avoid confusion in the code

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — DISMISSED** [review]
LGTM

**Review by jasonhenriquez — CHANGES_REQUESTED** [review]
**suggestion (blocking)**: ` --text-with-accent-color` (as primary) and `--text-with-main-color` (as secondary) for dark grey needs to be white (#FFFFFF) or another light color to resolve the issue of inaccessible color contrast with complementing text.

Screenshot_20231122_115203

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Comment by jasonhenriquez** [comment]
**nitpick:** Can you group the `text-with` colors with the rest on lines 240 and 537?

**Comment by jasonhenriquez** [comment]
You also need to set an explicit text color of black for the text color of light gray by adding its primary color to here:
Screenshot_20231128_151840

and the secondary color to here:
Screenshot_20231128_151944

**Comment by jasonhenriquez** [comment]
@fdarcey Thanks for adding the light gray changes! Could you also move these `text-with-main-color` and `text-with-accent-color` parts grouped with the other classes in the comma-delimited selectors that I mention [here](https://github.com/FreeTubeApp/FreeTube/pull/4293#issuecomment-1830737334) and [here](https://github.com/FreeTubeApp/FreeTube/pull/4293#issuecomment-1830765647)?

**Comment by fdarcey** [comment]
> @fdarcey Thanks for adding the light gray changes! Could you also move these `text-with-main-color` and `text-with-accent-color` parts grouped with the other classes in the comma-delimited selectors that I mention [here](https://github.com/FreeTubeApp/FreeTube/pull/4293#issuecomment-1830737334) and [here](https://github.com/FreeTubeApp/FreeTube/pull/4293#issuecomment-1830765647)? As in, for example, making this change:
> 
> ```
> .mainDarkGray,
> .mainRed,
> .mainPink,
> .mainPurple,
> .mainDeepPurple,
> .mainIndigo,
> .mainBlue,
> .mainLightBlue,
> .mainCyan,
> .mainTeal,
> .mainGreen {
>   --text-with-main-color: #FFFFFF;
>   --logo-icon-bar-color: url("../../_icons/iconWhiteSmall.png");
>   --logo-text-bar-color: url("../../_icons/textWhiteSmall.png");
> }
> ```
> 
> And doing the same for the proper places for secDarkGray, mainLightGray, and secLightGray.

I think I already did that, as shown here:

lightdarkmain
lightdarsec

Is there something wrong here?

**Comment by jasonhenriquez** [comment]
@fdarcey Oh yes, to clarify, didn't mean to say that you didn't have the code present. I'm just saying to do it in that manner I proposed, because doing it the other way also misses things like the `  --logo-icon-bar-color` and ` --logo-text-bar-color` that should be updated for these theme colors as well. That, and it allows us to more easily update the behaviors for these rules all at once if we need to change them or insert additional behaviors.

**Comment by fdarcey** [comment]
Sorry for vanishing a few weeks, the next update is coming soon.

**Comment by fdarcey** [comment]
I added the "--logo-icon-bar-color" and "--logo-text-bar-color" to the secondary grays to, I didn't see them associated with other secondary colors, but if they're not supposed to be there than it'd be simpler to remove them than add them.

**Comment by jasonhenriquez** [comment]
Thanks for your patience @fdarcey. Final asks:

Please just add `.mainLightGray` and `.mainDarkGray` to the top two sections instead of giving them their own sections. The reason is that code duplication can lead to updating issues down the road.
Screenshot_20240102_132156


Similarly, do the same for the secondary light and dark gray colors, adding them to the top two sections in this image & removing their own duplicated sections below. Those duplicated sections also modify the logo colors, which is incorrect.
Screenshot_20240102_132301

**Comment by fdarcey** [comment]
First of all, I'm sorry for forgetting about this. The schedule which once had a specific time I went to work on this project changed, and as a result other thinks took more of my mental space.

Secondly, I'm afraid that I don't understand my last task. I tried to fulfill it as best as I could interpret it, but I don't know if it was correct.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@jasonhenriquez 

> First of all, I'm sorry for forgetting about this. The schedule which once had a specific time I went to work on this project changed, and as a result other thinks took more of my mental space.
> 
> Secondly, I'm afraid that I don't understand my last task. I tried to fulfill it as best as I could interpret it, but I don't know if it was correct.

**Comment by jasonhenriquez** [comment]
Hi @fdarcey, I apologize for the communication troubles. I've created [this PR here](https://github.com/fdarcey/FreeTube/pull/1) with two commits for implementing the suggested changes.

File: src/renderer/themes.css [hunk 3]
```diff
@@ -566,6 +582,26 @@ it can be safely elided. This looks quite pleasant on this theme. */
 .secDeepOrange {
   --text-with-accent-color: #000000;
 }
+.secLightGray {
+  --text-with-accent-color: #000000;
+  --logo-icon-bar-color: url("../../_icons/iconBlackSmall.png");
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/themes.css:587]:
```suggestion
  --logo-icon-bar-color: url("../../_icons/iconBlackSmall.svg");
```

File: src/renderer/themes.css [hunk 4]
```diff
@@ -566,6 +582,26 @@ it can be safely elided. This looks quite pleasant on this theme. */
...
+  --logo-text-bar-color: url("../../_icons/textBlackSmall.png");
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/themes.css:588]:
```suggestion
  --logo-text-bar-color: url("../../_icons/textBlackSmall.svg");
```

File: src/renderer/themes.css [hunk 5]
```diff
@@ -566,6 +582,26 @@ it can be safely elided. This looks quite pleasant on this theme. */
...
+  --accent-color: #B1B1B1;
+  --accent-color-hover: #848482;
+  --accent-color-active: #7E7E7E;
+  --accent-color-light: #cfcfcf;
+  --accent-color-visited: #797979;
+}
+.secDarkGray {
+  --text-with-accent-color: #FFFFFF;
+  --logo-icon-bar-color: url("../../_icons/iconWhiteSmall.png");
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/themes.css:597]:
```suggestion
  --logo-icon-bar-color: url("../../_icons/iconWhiteSmall.svg");
```

File: src/renderer/themes.css [hunk 6]
```diff
@@ -566,6 +582,26 @@ it can be safely elided. This looks quite pleasant on this theme. */
...
+  --logo-text-bar-color: url("../../_icons/textWhiteSmall.png");
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/themes.css:598]:
```suggestion
  --logo-text-bar-color: url("../../_icons/textWhiteSmall.svg");
```

File: src/renderer/themes.css [hunk 7]
```diff
@@ -286,6 +286,22 @@ it can be safely elided. This looks quite pleasant on this theme. */
...
+  --text-with-main-color: #000000;
+  --logo-icon-bar-color: url("../../_icons/iconBlackSmall.png");
+  --logo-text-bar-color: url("../../_icons/textBlackSmall.png");
+  --primary-color: #B1B1B1;
+  --primary-color-hover: #848482;
+  --primary-color-active: #7E7E7E;
+}
+.mainDarkGray {
+  --text-with-main-color: #FFFFFF;
+  --logo-icon-bar-color: url("../../_icons/iconWhiteSmall.png");
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/themes.css:299]:
```suggestion
  --logo-icon-bar-color: url("../../_icons/iconWhiteSmall.svg");
```

File: src/renderer/themes.css [hunk 8]
```diff
@@ -286,6 +286,22 @@ it can be safely elided. This looks quite pleasant on this theme. */
...
+  --logo-text-bar-color: url("../../_icons/textWhiteSmall.png");
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/themes.css:300]:
```suggestion
  --logo-text-bar-color: url("../../_icons/textWhiteSmall.svg");
```

File: src/renderer/themes.css [hunk 9]
```diff
@@ -286,6 +286,22 @@ it can be safely elided. This looks quite pleasant on this theme. */
   --logo-icon-bar-color: url("../../_icons/iconBlackSmall.png");
   --logo-text-bar-color: url("../../_icons/textBlackSmall.png");
 }
+.mainLightGray {
+  --text-with-main-color: #000000;
+  --logo-icon-bar-color: url("../../_icons/iconBlackSmall.png");
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/themes.css:291]:
```suggestion
  --logo-icon-bar-color: url("../../_icons/iconBlackSmall.svg");
```

**Comment by jasonhenriquez** [comment]
To clarify, the above issues detected are merge conflict issues not detected by the GH action, and you should be able to resolve all of these with the version of the file shown in my commit above.

**Pull request closed without merging**